compiler/rustc_incremental/src/persist/work_product.rs

```diff
@@ -13,43 +13,58 @@ use std::path::Path;
 pub fn copy_cgu_workproduct_to_incr_comp_cache_dir(
     sess: &Session,
     cgu_name: &str,
-    path: &Path,
+    object_file_path: &Path,
+    dwarf_object_file_path: Option<&Path>,
```

Comment:
I did prefer an interface that allows arbitrary files. For cg_clif I want to be able to cache compiled inline assembly under the same workproduct so that I don't have to do a partial link of the main object file and the compiled inline assembly. Partial links are not supported on Windows AFAIK and are broken on macOS. This is one of the main blockers for fully functioning inline assembly support in cg_clif.

Reply:
I decided to add to `WorkProduct` because it looked like the rest of the compiler was assuming a single `WorkProduct` per CGU.

I could change this pull request to just store a `Vec<String>` in the `WorkProduct`, or I could make it easier for multiple `WorkProduct`s to be associated with a single CGU. In either case, we'd probably want some way of working out what a `WorkProduct` has without just working it out from filenames? Or I could leave it as-is in the pull request and this mechanism can be made more generic in a follow-up (which I'd be happy to do).

Reply:
> I could change this pull request to just store a Vec<String> in the WorkProduct

This or having `HashMap<String, String>` would be my preferred interface.

> or I could make it easier for multiple WorkProducts to be associated with a single CGU.

This makes less sense to me as all compiled results of a cgu are changed together and a `WorkProduct` is the smallest unit of incremental compilation for codegen.

Reply:
I've updated this pull request so that `WorkProduct` contains a `HashMap<String, String>`

Reply:
https://github.com/bjorn3/rustc_codegen_cranelift/commit/e45f6000a0bd46d4b7580db59c86f3d30adbc270 implemented what I was talking about above.